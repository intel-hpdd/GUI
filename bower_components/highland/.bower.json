{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}