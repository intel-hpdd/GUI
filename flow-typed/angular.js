// @flow

declare module angular {
  declare class $scope {
    $new():$scope;
    $digest():void;
  }
  declare type $scopeT = $scope;
  declare class Module {
    provider(name:string, providerType:Function):Module;
    factory(name:string, providerFunction:Function):Module;
    controller(name:string, constructor:Function):Module;
    service(name:string, constructor:Function):Module;
    config(configFn:Function):Module;
    run(initializationFn:Function):Module;
    directive(name:string, directiveFactory:Function):Module;
    component(name:string, options:Object):Module;
    filter(name:string, filterFactory:Function):Module;
    value(name:string, object:any):Module;
    constant(name:string, object:any):Module;
    name:string;
  }
  declare type $compileT = (el:(string | HTMLElement)) => (scope:$scope) => HTMLElement[];
  declare module.exports: {
    module(name:string, dependencies: string[]):Module;
    bootstrap(element:(HTMLElement | Document), modules:string[], config:Object):void;
    merge(...rest:Object[]):Object;
    element(el:(string | HTMLElement)): {
      clone: () => HTMLElement
    };
  }
}
