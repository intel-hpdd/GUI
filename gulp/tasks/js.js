//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
