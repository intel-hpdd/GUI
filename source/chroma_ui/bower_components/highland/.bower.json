{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}