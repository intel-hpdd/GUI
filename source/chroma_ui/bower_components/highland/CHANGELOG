Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
