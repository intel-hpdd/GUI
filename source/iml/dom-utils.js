// @flow

type QuerySelector<T: HTMLElement> = (HTMLElement | Document, string) => T;

export const querySelector: QuerySelector<*> = (
  elm: HTMLElement | Document,
  sel: string
) => {
  const selected = elm.querySelector(sel);

  if (!selected)
    throw new Error(
      `Could not find selector ${sel} for element ${elm.nodeName}`
    );

  return selected;
};
