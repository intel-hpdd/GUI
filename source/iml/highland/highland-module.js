// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
