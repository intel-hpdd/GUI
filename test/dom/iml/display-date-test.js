// @flow

import { querySelector } from "../../../source/iml/dom-utils.js";
import angular from "../../angular-mock-setup.js";
import highland from "highland";

import type { HighlandStreamT } from "highland";

type UtcStream = HighlandStreamT<{ isUtc: boolean }>;
interface MockMoment {
  utc: Function;
}

describe("Display Date Component", () => {
  let root: HTMLElement,
    displayDate,
    dateElement: HTMLElement,
    $scope: $scope,
    $compile,
    template: string,
    tzPicker$: UtcStream,
    tzPickerB: () => UtcStream,
    mockMoment: MockMoment,
    mockMomentFormat,
    mockMomentUtc,
    mockMomentUtcFormat;

  beforeEach(() => {
    tzPicker$ = highland();
    tzPickerB = jest.fn(() => tzPicker$);

    mockMomentUtcFormat = jest.fn(() => "2018-07-01 12:30:00");
    mockMomentUtc = jest.fn(() => ({
      format: mockMomentUtcFormat
    }));

    mockMomentFormat = jest.fn(() => "2018-07-01 08:30:00");
    mockMoment = (jest.fn(() => ({
      format: mockMomentFormat
    })): any);
    mockMoment.utc = mockMomentUtc;

    jest.mock("moment", () => mockMoment);
    displayDate = require("../../../source/iml/display-date.js").displayDateComponent;
  });

  beforeEach(
    angular.mock.module($compileProvider => {
      $compileProvider.component("displayDate", displayDate);
    })
  );

  beforeEach(
    angular.mock.inject((_$compile_, $rootScope) => {
      $scope = $rootScope.$new();
      $scope.tzPickerB = tzPickerB;
      $scope.datetime = "2018-07-01T12:30:00.0000+00:00";
      $compile = _$compile_;
      template = '<display-date tz-picker-b="tzPickerB" datetime="datetime"></tz-picker>';
    })
  );

  beforeEach(() => {
    tzPicker$.write({ isUtc: false });

    root = $compile(template)($scope)[0];
    querySelector(document, "body").appendChild(root);
    dateElement = querySelector(root, ".date");
  });

  afterEach(() => {
    querySelector(document, "body").removeChild(root);
  });

  describe("local time", () => {
    it("should create markup", () => {
      expect(dateElement).toMatchSnapshot();
    });

    it("should pass the ISO date to moment", () => {
      expect(mockMoment).toHaveBeenCalledOnceWith("2018-07-01T12:30:00.0000+00:00");
    });

    it("should format the date", () => {
      expect(mockMomentFormat).toHaveBeenCalledOnceWith("YYYY-MM-DD HH:mm:ss");
    });
  });

  describe("utc", () => {
    beforeEach(() => {
      tzPicker$.write({ isUtc: true });
    });

    it("should create the markup", () => {
      expect(dateElement).toMatchSnapshot();
    });

    it("should pass the ISO date to moment", () => {
      expect(mockMomentUtc).toHaveBeenCalledOnceWith("2018-07-01T12:30:00.0000+00:00");
    });

    it("should format the date", () => {
      expect(mockMomentUtcFormat).toHaveBeenCalledOnceWith("YYYY-MM-DD HH:mm:ss");
    });
  });

  describe("removing", () => {
    it("should no longer render the element", () => {
      $scope.$destroy();
      const removedDateDisplay = root.querySelector(".date");
      expect(removedDateDisplay).toBeNull();
    });
  });
});
