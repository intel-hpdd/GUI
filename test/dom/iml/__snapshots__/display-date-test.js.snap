// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Date Component local time should create markup 1`] = `
<span
  class="date"
>
  <span>
    2018-07-01 08:30:00
  </span>
</span>
`;

exports[`Display Date Component utc should create the markup 1`] = `
<span
  class="date"
>
  <span>
    2018-07-01 12:30:00
  </span>
</span>
`;
